### PR TITLE
Improve PDF/A validation to include more robust reporting on failures and update PDF validation against '1a' and '1b' flavors

### LINKED ISSUE #388: Improve PDF/A validation to include more robust reporting on failures
## 💪 Motivation
...so that I can know why my PDF is failing validation.

## 📖 Additional Details

Refs #382 where "supposed" PDF/A files are failing but the user does not know why. Let's output the verapdf report so folks can try to dissect it.

## ⚖️ Acceptance Criteria
**Given** PDS4 label `document.xml` referencing a PDF `document.pdf` that is not PDF/A compliant
**When I perform** validation with content validation enabled (default functionality)
**Then I expect** validate to fail when validating `document.xml` and the validation report should refer to a file on the file system where the verapdf validation report will be output.

## ⚙️ Engineering Details

- [x] update software to output verapdf report to file on file system (make sure it is OS agnostic)
- [x] update validate documentation:
    - [x] new section noting we are validating PDF/A
    - [x] note that we are using verapdf with link to the website
    - [x] note that the referenced output reports are directly from verapdf. for more information on what PDF/A-1A conformance really means, see [PDFA.org documentation](https://www.pdfa.org/accessibility-what-pdf-a-1a-really-means/)

### LINKED ISSUE #412: Update PDF validation to check against flavour specified in PDF metadata
1. Check that the flavour is 1a or 1b
2. Validate the PDF is actually valid against that version

Per Jesse S:

PDFUtil.java already uses parser.getFlavour(), but passes it into createValidator: https://github.com/NASA-PDS/validate/blob/0e9984ea92e59d34dd6d74b298ac247ddf1f382b/src/main/java/gov/nasa/pds/tools/util/PDFUtil.java#L75

Meanwhile, PDFAFlavour is just a java enum, so it should be very easy to check against: https://github.com/veraPDF/veraPDF-library/blob/6a376fbd4faf4818c6123a5bdc4ce9ea03f1db32/core/src/main/java/org/verapdf/pdfa/flavours/PDFAFlavour.java

## Related Requirements
🦄 https://github.com/NASA-PDS/validate/issues/164

### PR DESCRIPTION
**Pull request by qchaupds**
Closes https://github.com/nasa-pds/validate/issues/388 Improve PDF/A validation to include more robust reporting on failures
Closes https://github.com/nasa-pds/validate/issues/412 Update PDF validation to check against flavour specified in PDF metadata

A new requirement to have a more robust reporting of PDF validating so an external file was created when a specific flavor parser was used
to validate a PDF document.

Tested in DEV:

Case 1:
Running validate against a bundle with PDF that is not valid.

```
validate -R pds4.bundle  -r report_github328_bundle_invalid.json   -s json -t src/test/resources/github328/invalid/bundle_misc.xml 
```

There should be 2 .error files created.  One for '1a' and one for '1b' flavor.

```

% ls -1ctr TITAN*.error 

TITAN_NEUTRALS_UG_ITAR_SEP2011.PDF.1a.error
TITAN_NEUTRALS_UG_ITAR_SEP2011.PDF.1b.error
```

### CODE REVIEW

File: src/main/java/gov/nasa/pds/tools/util/PDFUtil.java [hunk 1]
```diff
@@ -38,14 +52,82 @@ public URL getTarget() {
...
+      // Write to an external file with ".error" appended to file name in user's default directory the content of result.
+
+      // Build the external filename and save it for other to access.
+      this.externalErrorFilename = System.getProperty("user.dir") + File.separator + FilenameUtils.getName(uri.getPath()) + "." + flavor + ".error";
+
+      LOG.debug("writeErrorToFile:uri,this.externalErrorFilename {},{}",uri,this.externalErrorFilename);
+
+      FileWriter myWriter = null;
+      try {
+          myWriter = new FileWriter(this.externalErrorFilename);
```
[the remark this answers is not in the document]
Comment (qchaupds) [reply src/main/java/gov/nasa/pds/tools/util/PDFUtil.java:66]: @sonatype-lift ignore

[the remark this answers is not in the document]
Comment (qchaupds) [reply src/main/java/gov/nasa/pds/tools/util/PDFUtil.java:66]: @sonatype-lift ignore

File: src/main/java/gov/nasa/pds/tools/util/LidVid.java [hunk 2]
```diff
@@ -43,21 +43,32 @@
...
-      String largestVersion = null;
-      if (vidsList.size() == 0) {
-          return(largestVersion);
+    HashMap<String,String> largestVersion = new HashMap<String,String>(); // A HashMap array containing a map of logical identifier and its latest version.
+
+    LOG.debug("getLatestVersion:lidvidsMap.size {}",lidvidsMap.size());
+
+    // Loop through each logical_identifier in lidvidsMap and find the largest version for that logical_identifier.
+    for (String lid : lidvidsMap.keySet()) {
+      ArrayList<String> vidsList = lidvidsMap.get(lid);  // The value in HashMap is an array of version_id list.
```
[the remark this answers is not in the document]
Comment (qchaupds) [reply src/main/java/gov/nasa/pds/tools/util/LidVid.java]: @sonatype-lift ignore

File: src/main/java/gov/nasa/pds/tools/util/LidVid.java [hunk 3]
```diff
@@ -127,32 +152,60 @@ public static String getLatestVersion(List<String> vidsList) {
...
       // The target with the largest version is the desired one to return.
       // Because largestVersion can be null if the list of targets is empty, it must be checked.
-      if (largestVersion != null) {
-          reducedKids.add(vidsTable.get(largestVersion));
-          LOG.debug("reduceToLatestTargetOnly:vidsTable.get(largestVersion) {}",vidsTable.get(largestVersion));
+      if (largestVersion != null && largestVersion.size() > 0) {
+          String fullyQualifiedLogicalIdentifierVersion = null;  // Also known as lidvid.
+          for (String singleLogicalIdentifier  : largestVersion.keySet()) {
+              // To build the key to vidsTable, it is necessary to add the logical identifier to "::" and then the version_id
+              fullyQualifiedLogicalIdentifierVersion = singleLogicalIdentifier + "::" + largestVersion.get(singleLogicalIdentifier);
```
[the remark this answers is not in the document]
Comment (qchaupds) [reply src/main/java/gov/nasa/pds/tools/util/LidVid.java]: @sonatype-lift ignore

File: src/main/java/gov/nasa/pds/tools/util/LidVid.java [hunk 4]
```diff
@@ -43,21 +43,32 @@
...
-      if (vidsList.size() == 0) {
-          return(largestVersion);
+    HashMap<String,String> largestVersion = new HashMap<String,String>(); // A HashMap array containing a map of logical identifier and its latest version.
+
+    LOG.debug("getLatestVersion:lidvidsMap.size {}",lidvidsMap.size());
+
+    // Loop through each logical_identifier in lidvidsMap and find the largest version for that logical_identifier.
+    for (String lid : lidvidsMap.keySet()) {
+      ArrayList<String> vidsList = lidvidsMap.get(lid);  // The value in HashMap is an array of version_id list.
+      LOG.debug("getLatestVersion:vidsList,vidsList.size {},{}",vidsList,vidsList.size());
```
[the remark this answers is not in the document]
Comment (qchaupds) [reply src/main/java/gov/nasa/pds/tools/util/LidVid.java]: @sonatype-lift ignore

**Comment by jordanpadams** [comment]
@qchaupds can we improve the title for the PR when you have a chance also.

**Comment by qchaupds** [comment]
@jordanpadams How to change the title?  I didn't noticed that the title is just the name of the branch for the last few PRs.  It used to be the commit message.  Nevermind, I found the "edit" button on the right of the title.  Was expecting some ... symbol.

**Comment by qchaupds** [comment]
@jordanpadams Also, noticed there is a conflict with the feature file.  We want to add any tests that are new.

**Comment by jordanpadams** [comment]
@qchaupds click the edit button to the right of the issue.

For resolving conflicts click the resolve conflicts button and you can resolve in the browser.

**Comment by qchaupds** [comment]
@jordanpadams The conflict seems to have merged.  Please double check when you do your final merge.

**Comment by jordanpadams** [comment]
@qchaupds I made some updates here: 40f010a . let me know if that looks ok and we can go with a merge.

**Comment by qchaupds** [comment]
@jordanpadams Thanks for doing the updates.   Your changes were easier to read.  Please do the merge.

**Review by jordanpadams — APPROVED** [review]
per comment from @qchaupds

**Pull request merged**